test/aws/rails/mailer_test.rb

```diff
@@ -25,9 +35,11 @@ def test_settings_method
 
       def test_deliver
         message = sample_message
-        resp = @mailer.deliver!(message)
-        assert_equal resp.context.params[:raw_message][:data].to_s, message.to_s
-        assert_equal resp.context.params[:destinations], message.destinations
+        data = @mailer.deliver!(message).context.params
+        body = data[:raw_message][:data].to_s
+        body.gsub!("\r\nHallo", "ses-message-id: #{message_id}\r\n\r\nHallo")
```

Comment:
I'd actually like the test to mimic the behavior pattern you showed in the PR description. Perhaps retrieve the message body and message headers like a user would?

Reply:
Good idea, that’s pushed.